File: TableGames.Games/playerHand.js
define(['knockout', 'game!card'], function(ko, Card) {

    function PlayerHand(playerHandState) {
        var self = this;
        self.playerName = playerHandState.playerName;
        self.cards = ko.observableArray(playerHandState.cards.map(function(cardState) { return new Card(cardState); }));

        self.updateCards = function(cardsState) {
            cardsState.forEach(function(cardState, index) {
                var card = self.cards()[index];
                if (card) {
                    card.suit(cardState.suit);
                    card.rank(cardState.rank);
                    card.isExposed(cardState.isExposed);
                    card.isHighlighted(cardState.isHighlighted);
                } else {
                    self.cards.push(new Card(cardState));
                }
            });
        };
    }

    return PlayerHand;
});

File: TableGames.Games/PlayerHand.cs
using System.Collections.Generic;
using System.Linq;
using TableGames.Domain;

namespace TableGames.Games
{
    public class PlayerHand
    {
        public Player Player { get; set; }
        public List<Card> Cards { get; set; }

        public PlayerHand(Player player) {
            Player = player;
            Cards = new List<Card>();
        }

        public void ExposeCards() {
            Cards.ForEach(d => d.IsExposed = true);
        }

        public virtual object ToClient() {
            return new {
                playerName = Player.Name,
                cards = Cards.Select(d => d.ToClient()),
            };
        }
    }
}